kms_log/main.tf

```diff
@@ -125,7 +124,7 @@ data "aws_iam_policy_document" "sqs_kms_ct_events_policy" {
             test = "StringLike"
             variable = "aws:SourceArn"
             values = [
-                "${aws_cloudwatch_event_rule.decrypt.arn}"
+                "arn:aws:events:${var.region}:${data.aws_caller_identity.current.account_id}:rule/${var.env_name}-decryption-events"
```

Comment:
Alternatively you can use the first element of concat * terraform workaround for accessing resources that aren't guaranteed to exist.

Reply:
Tried the concat and coalesce and it failed.  Added a local variable for the name so there is only one place to change.